### PR TITLE
add getCancellationReason endpoint to ManagmentOrderController

### PR DESCRIPTION
**Pull request by KharchenkoV**
dev
## Sumary of issue:
Need to add new endpoint for getting cancellation reason and comment

bug
[#5479](https://github.com/ita-social-projects/GreenCity/issues/5479)

## Summary of change
ManagmentOrderController  added getOrderCancellationReason method
ManagmentOrderController made test for getOrderCancellationReason method
UBSManagmentService added getOrderCancellationReason method
UBSManagmentServiceImpl implemented getOrderCancellationReason method
UBSManagmentServiceImplTest made test for getOrderCancellationReason method
SecurityConfig added  "/ubs/management/get-order-cancellation-reason/{id}" endpoint 

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: core/src/test/java/greencity/controller/ManagementOrderControllerTest.java [hunk 1]
```diff
@@ -455,4 +455,10 @@ void addPaymentDiscountTest() throws Exception {
             .andExpect(status().isCreated());
 
     }
+
+    @Test
+    void getOrderCancellationReason() throws Exception {
+        this.mockMvc.perform(get(ubsLink + "/get-order-cancellation-reason" + "/{id}", 1L))
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/controller/ManagementOrderControllerTest.java:461]: please add `verify` that` ubsManagementService.getOrderCancellationReason` is called with expected params

File: service-api/src/main/java/greencity/service/ubs/UBSManagementService.java [hunk 2]
```diff
@@ -281,4 +281,11 @@ void updateOrderAdminPageInfo(UpdateOrderPageAdminDto updateOrderPageAdminDto, L
      * @author Anatolii Shapiro.
      */
     void updateOrderStatusToExpected();
+
+    /**
+     * Method returns cancellation reason and comment to order.
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/service/ubs/UBSManagementService.java:286]: please add `id` and return type to javadoc

File: service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java [hunk 3]
```diff
@@ -1691,4 +1692,14 @@ public void updateOrderStatusToExpected() {
             OrderStatus.ON_THE_ROUTE.name(),
             LocalDate.now());
     }
+
+    @Override
+    public OrderCancellationReasonDto getOrderCancellationReason(Long orderId) {
+        Order order = orderRepository.findById(orderId)
+            .orElseThrow(() -> new NotFoundException(ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST));
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java:1699]: please add test if` NotFoundException(ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST)`

File: service/src/test/java/greencity/service/ubs/UBSManagementServiceImplTest.java [hunk 4]
```diff
@@ -2171,4 +2172,14 @@ void updateOrderExportDetailsWhenDeliverToIsNull() {
         verify(orderRepository, times(1)).save(order);
         verify(eventService, times(1)).saveEvent(expectedHistoryEvent, employee.getEmail(), order);
     }
+
+    @Test
+    void getOrderCancellationReasonTest() {
+        OrderCancellationReasonDto cancellationReasonDto = ModelUtils.getCancellationDto();
+        Order order = ModelUtils.getOrderTest();
+        when(orderRepository.findById(anyLong())).thenReturn(Optional.ofNullable(order));
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/ubs/UBSManagementServiceImplTest.java]: please do not use `anyLong()`
It is better to use expected `orderId`
And please add `verify`

File: core/src/main/java/greencity/controller/ManagementOrderController.java [hunk 5]
```diff
@@ -967,4 +968,25 @@ public ResponseEntity<AddBonusesToUserDto> addBonusesToUser(@PathVariable(name =
...
+     * @return {@link OrderCancellationReasonDto}.
+     * @author Kharchenko Volodymyr
+     */
+    @ApiOperation(value = "Get order cancellation reason")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK, response = OrderCancellationReasonDto.class),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+        @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/ManagementOrderController.java:984]: I see that we can have next responses:

- `200` - success
- `401` - unauthorized user
- `403` - used does not have the required role
- `404` - order with `orderId` does not exist

When can we have `400` response?

File: core/src/main/java/greencity/controller/ManagementOrderController.java [hunk 6]
```diff
@@ -967,4 +968,25 @@ public ResponseEntity<AddBonusesToUserDto> addBonusesToUser(@PathVariable(name =
...
+    })
+    @GetMapping("/get-order-cancellation-reason/{id}")
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/ManagementOrderController.java:986]: I see that there is a similar endpoint in `OrderController`
```
 @GetMapping("/order/{id}/cancellation")
    public ResponseEntity<OrderCancellationReasonDto> getCancellationReason(
```
Why do we need one more?
can we modify the existing endpoint?

**Review by RomaMocherniuk — CHANGES_REQUESTED** [review]
Left some comments

File: core/src/test/java/greencity/controller/ManagementOrderControllerTest.java (hunk 1 above)
Comment (KharchenkoV) [reply core/src/test/java/greencity/controller/ManagementOrderControllerTest.java:461]: done

File: service-api/src/main/java/greencity/service/ubs/UBSManagementService.java (hunk 2 above)
Comment (KharchenkoV) [reply service-api/src/main/java/greencity/service/ubs/UBSManagementService.java:286]: done

File: service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java (hunk 3 above)
Comment (KharchenkoV) [reply service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java:1699]: done

File: service/src/test/java/greencity/service/ubs/UBSManagementServiceImplTest.java (hunk 4 above)
Comment (KharchenkoV) [reply service/src/test/java/greencity/service/ubs/UBSManagementServiceImplTest.java]: done

File: core/src/main/java/greencity/controller/ManagementOrderController.java (hunk 6 above)
Comment (KharchenkoV) [reply core/src/main/java/greencity/controller/ManagementOrderController.java:986]: We need another one only for admin

**Review by RomaMocherniuk — APPROVED** [review]

**Review by LiliaMokhnatska — APPROVED** [review]

**Review by ABbondar — APPROVED** [review]

**Review by MaksymGolik — APPROVED** [review]

**Pull request merged**